src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs

```diff
@@ -57,6 +58,67 @@ internal AnnotatedCodeLocationVersionOne CreateAnnotatedCodeLocation(Location v2
             return annotatedCodeLocation;
         }
 
+        internal AnnotatedCodeLocationVersionOne CreateAnnotatedCodeLocation(ThreadFlowLocation v2ThreadFlowLocation)
+        {
+            AnnotatedCodeLocationVersionOne annotatedCodeLocation = null;
+
+            if (v2ThreadFlowLocation != null)
+            {
+                annotatedCodeLocation = CreateAnnotatedCodeLocation(v2ThreadFlowLocation.Location);
+                annotatedCodeLocation = annotatedCodeLocation ?? new AnnotatedCodeLocationVersionOne();
+
+                annotatedCodeLocation.Importance = Utilities.CreateAnnotatedCodeLocationImportance(v2ThreadFlowLocation.Importance);
+                annotatedCodeLocation.Module = v2ThreadFlowLocation.Module;
+                annotatedCodeLocation.Properties = v2ThreadFlowLocation.Properties;
+                annotatedCodeLocation.State = v2ThreadFlowLocation.State;
+                annotatedCodeLocation.Step = v2ThreadFlowLocation.ExecutionOrder;
```

Comment:
Bug: `step` is not the same as `executionOrder`. `step` starts at 1 and increases by 1 for each location in a `threadFlow`. `executionOrder` is optional; it's only needed when you need to specify the ordering of locations on different threads. I gave an example in a comment on the previous review #947.

Reply:
But wait! We've decided not to proceed with the down-conversion from v2 to v1 at all.

Reply:
Where is the up-conversion code for translating v1 to v2 codeFlows? Did you already merge that?

Reply:
Yeah this is the transformer we're going to remove.

Reply:
Maybe I'm misunderstanding the plan

Reply:
The plan is:
1. Finish work on the v1-to-v2 codeFlow transformer. I don't see any of that code here.
2. Cease work on the v2-to-v1 transformer.
3. In a subsequent PR, _remove_ the v2-to-v1 transformer entirely.

So I'm confused. Have we already merged the v1-to-v2 codeFlow transformer, and I forgot?

Reply:
1. I have this in a separate change.
2. This is why the above code isn't finished.

Reply:
But we shouldn't merge this code at all.

Reply:
OK I'll remove it.

Reply:
Cool.

Reply:
Now, when you say the v1-to-v2 conversion (which we _do_ want) is "in a separate change", do you mean it's in a previous PR that's already merged, or in a separate branch in your enlistment that you haven't pushed a PR for yet? Or something else?

Reply:
It's a change I made on Friday that I haven't PRed yet. I wanted this change in because I'm moving things around in the sarif.unittests project.

Reply:
gentlemen, can we please discuss the v2 -> v1 issue before we take any action?

Reply:
Sure. I thought abandoning it was a baked plan. Want to Skype?

Reply:
let's talk in person. someone was just asking today for a Binskim v2 -> v1 transformer. I'm guessing we can do this. and so i wonder whether we want to throw out everything just because we can't transform everything. i.e., what if this code did a good job with what it can do and ignored the rest or errored out? people would then get benefit of the working limited functionality.

make sense?

Reply:
That does make sense. And binskim doesn't use codeflows, right?

Do you want to talk IRL or on Skype? I can come in tomorrow if (absolutely :-)) necessary.

Reply:
If you guys are meeting, please include me as well.